src/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java

```diff
@@ -738,18 +738,22 @@ Element createParameterElement( final ParameterDefinitionEntry parameter,
           final String namespace = namespaces[ i ];
           final String[] attributeNames = parameter.getParameterAttributeNames( namespace );
           for ( final String attributeName : attributeNames ) {
-            final String attributeValue = parameter.getParameterAttribute( namespace, attributeName, parameterContext );
-            // expecting: label, parameter-render-type, parameter-layout
-            // but others possible as well, so we set them all
-            final Element attributeElement = document.createElement( "attribute" ); // NON-NLS
-            attributeElement.setAttribute( "namespace", namespace ); // NON-NLS
-            attributeElement.setAttribute( "name", attributeName ); // NON-NLS
-            attributeElement.setAttribute( "value", attributeValue ); // NON-NLS
-
-            parameterElement.appendChild( attributeElement );
+            final String attributeValue =
```

Comment:
Yes, it is more than OK to send those attributes. We call it a feature, as it allows OEMs to add their own attributes to parameters to drive their own logic in their parameter UIs.

And yes, that class could need a clean up. The drawback of using DOM is that this code is verbose like hell. But anything we can do to keep the new code as slim, simple and easy to read at least wont make the situation worse.